riscv-page-tables/src/page_table.rs

```diff
@@ -465,6 +460,10 @@ pub trait PagingMode {
     /// Calculates the number of PTE pages that are needed to map all pages for `num_pages` mapped
     /// pages for this type of page table.
     fn max_pte_pages(num_pages: u64) -> u64;
+
+    /// Used to calculate the number of PTE pages needed if all the pages are
+    /// not 4k
+    fn max_huge_pte_pages(num_pages: u64, page_size: PageSize) -> u64;
```

Comment:
This and `max_pte_pages()` should probably take a length rather than a number of (assumed to be) 4kB pages.

Also, `max_huge_pte_pages()` works just fine with a non-4kB size; `max_pte_pages_with_size()` would be a better name. Then you could do something like:

```
fn max_pte_pages_with_size(len: u64, page_size: PageSize) -> u64;

fn max_pte_pages(len: u64) -> u64 {
    Self::max_pte_pages_with_size(len, PageSize::Size4k)
}
```

(BTW -- where is this function used? I don't see it called anywhere in this patch.)

Reply:
Got rid of the function. May come back later.